Feature exclude tactic name

frederikb96 opened the pull request:
<!--
Thank you for your interest in and contributing to Detection Rules!
There are a few simple things to check before submitting your pull request
that can help with the review process. You should delete these items
from your submission, but they are here to help bring them to your attention.
-->
# Pull Request



<!--
  Add Related Issues / PRs for context. Eg:
    Related to elastic/repo#999
    Resolves #123
  If there is no issue link, take extra care to write a clear summary and label the PR just as you would label an issue to give additional context to reviewers.
-->

## Summary - What I changed

<!--
  Summarize your PR. Animated gifs are 💯. Code snippets are ⚡️. Examples & screenshots are 🔥
-->

1. You use the repository for DaC and have a custom rule folder for your company rules
2. You export rules from kibana to the folder via the repos CLI
3. The name of the toml files for the rules are all prefixed with the tactic name but you dont want this since you have a clear naming convention for all your rules and the toml file name should be just the rule name (no tactic prefix)
4. So I added this new flag: "--no-tactic-filename"
5. With this the toml files are simply named like the rule name, nice!

The adjustment was straightforward, check the code diff :)

## How To Test

<!--
  Some examples of what you could include here are:
  * Links to GitHub action results for CI test improvements
  * Sample data before/after screenshots (or short videos showing how something works)
  * Copy/pasted commands and output from the testing you did in your local terminal window
  * If tests run in GitHub, you can 🪁or 🔱, respectively, to indicate tests will run in CI
  * Query used in your stack to verify the change
-->

1. Export rules via CLI
```
python -m detection_rules kibana export-rules -d custom-rules
```
2. And now use the new flag
```
python -m detection_rules kibana export-rules -d custom-rules --no-tactic-filename
```
Filenames are just the rule name now :rocket: 


## Checklist

<!-- Delete any items that are not applicable to this PR. -->

- [x] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `maintenance`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [x] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [x] Secret and sensitive material has been managed correctly
- [x] Automated testing was updated or added to match the most common scenarios
- [x] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

eric-forte-elastic commented:
:wave: Thanks for the PR and suggestion! A few other things that we should think about and/or address is that we have a unit test that our users would want to bypass if this approach. This can be done if adding the following bypass line to the `test_config.yaml`
```
- tests.test_all_rules.TestRuleFiles.test_rule_file_name_tactic
```

Additionally, even with this test we have a warning message as part of the loading of rules in a RuleCollection that will warn that the name does not match. 
E.g.
```
WARNING: Rule path does not match required path: signin_console_login_no_mfa.toml != initial_access_aws_signin_single_factor_console_login_with_federated_user.toml
```

To test this, you can use any of the commands that load our rules into a RulesCollection, but for DaC testing specifically you can use something like the following:
```
 python -m detection_rules kibana import-rules -d new_dac_demo/rules/ -e -ac -o
```

You may want to consider adding this as a configuration option that will disable this warning if used.

Also we should add this functionality to `import-rules-to-repo` to have feature parity.

eric-forte-elastic commented:
@frederikb96 Happy to, this is a great PR :rocket: 

A few notes, could you bump the patch version in the pyproject.toml? And yes you are correct with the parity that field is not available so you would already have achived parity :slightly_smiling_face:, so you can remove the argument from import-rules-to-repo. One last thing, if you could add the functionality to disable the warning message that would be great! (see below for where it checks for the naming convention) 
https://github.com/elastic/detection-rules/blob/753e8d8200f4d8b65651d9b7cd8bcd8492325d04/detection_rules/cli_utils.py#L103-L106 

Also, to check for linting issues (there appear to be some) you can use `python -m flake8 tests detection_rules --ignore D203,N815 --max-line-length 120` to help pass the unit tests :slightly_smiling_face:

frederikb96 commented:
@eric-forte-elastic Thanks again, that was a great hint. Way nicer with the config option :+1: 

I adjusted again:
1. Version adjusted
2. Config option added in config.py, the _config.yaml as comment, the kibana export function and the kibana import function.
3. The warning is not displayed anymore when using the config or the flag option during kibana import-rules
4. Used the lint command and adjusted the code lines

frederikb96 commented:
@eric-forte-elastic Anything else or would it be ready to merge? :)

eric-forte-elastic commented:
> @eric-forte-elastic Anything else or would it be ready to merge? :)

This looks really good, will run some tests and put results here, but otherwise looks good to me :+1:

eric-forte-elastic commented:
@frederikb96 I think you may need to update the pyproject toml for the required patch version bump.

Mikaayenson reviewed (approved):
👋 Thanks for the contribution!

frederikb96 commented:
> This looks very good! Going to go head and merge this on your behalf @frederikb96 so that we can get this in without having to go back and forth bumping the toml version again 😂

Yes nice, thanks :blush: :smile:
